### PR TITLE
Update Python bindings, port tests to Pytest

### PR DESCRIPTION
**Pull request by Luthaf**
Part of #189

----
:books: Documentation preview :books:: https://rascaline--192.org.readthedocs.build/en/192/

### CODE REVIEW

**Comment by Luthaf** [comment]
To be merged after #191

File: python/tests/log.py [hunk 1]
```diff
@@ -8,70 +8,66 @@
...
-            "log-test-warn: this is a test warning message"
-        )
-        event = (RASCAL_LOG_LEVEL_WARN, message)
-        self.assertTrue(event in recorded_events)
+    calculator = DummyCalculator(
+        cutoff=3.2,
+        delta=0,
+        name="log-test-info: test info message",
+    )
+    _ = calculator.compute(SystemForTests())
```
Comment (PicoCentauri) [inline python/tests/log.py:35]:
```suggestion
    calculator.compute(SystemForTests())
```

File: python/tests/log.py [hunk 2]
```diff
@@ -8,70 +8,66 @@
...
 
-        set_logging_callback(record_log_events)
 
-        calculator = DummyCalculator(
-            cutoff=3.2,
-            delta=0,
-            name="log-test-info: test info message",
-        )
-        _ = calculator.compute(TestSystem())
+def test_log_message():
```
Comment (PicoCentauri) [inline python/tests/log.py:22]: There also exists an option in pytest to collect the log. This is fine here but just for the future.

https://docs.pytest.org/en/6.2.x/logging.html

File: python/tests/systems/ase.py [hunk 3]
```diff
@@ -1,154 +1,147 @@
...
+    atoms = ase.Atoms("C", positions=[(0, 0, 0)])
+    atoms.pbc = [True, True, True]
+
+    message = (
+        "periodic boundary conditions are enabled, but the cell "
+        "matrix is zero everywhere. You should set pbc to `False`, "
+        "or the cell to its value."
+    )
+    with pytest.raises(ValueError, match=message):
+        _ = AseSystem(atoms)
```
Comment (PicoCentauri) [inline python/tests/systems/ase.py:114]:
```suggestion
        AseSystem(atoms)
```

File: python/tests/systems/ase.py [hunk 4]
```diff
@@ -1,154 +1,147 @@
...
+
+    atoms.cell = [[0, 0, 0], [0, 0, 0], [0, 0, 0]]
+    atoms.pbc = True
+
+    with pytest.raises(ValueError, match=message):
+        _ = AseSystem(atoms)
```
Comment (PicoCentauri) [inline python/tests/systems/ase.py:120]:
```suggestion
        AseSystem(atoms)
```

File: python/tests/systems/ase.py [hunk 5]
```diff
@@ -1,154 +1,147 @@
...
+    atoms = ase.Atoms("C", positions=[(0, 0, 0)])
+
+    atoms.cell = [[10, 0, 0], [0, 10, 0], [0, 0, 10]]
+    atoms.pbc = [True, True, False]
+
+    message = (
+        "different periodic boundary conditions on different axis " "are not supported"
+    )
+    with pytest.raises(ValueError, match=message):
+        _ = AseSystem(atoms)
```
Comment (PicoCentauri) [inline python/tests/systems/ase.py:133]:
```suggestion
        AseSystem(atoms)
```

File: tox.ini [hunk 6]
```diff
@@ -8,30 +9,41 @@ envlist =
...
-passenv =
-    RASCALINE_BUILD_TYPE
-    RUSTUP_HOME
-    CARGO*
+package = external
+package_env = build-rascaline
+passenv = *
 
+rascaline-dependencies =
+    equistore-core @ https://github.com/lab-cosmo/equistore/archive/3bced4c.zip\#subdirectory=python/equistore-core
```
Comment (PicoCentauri) [inline tox.ini:17]: Remind me why do we need this here? Shouldn't this be come automatically from the package dependencies?

**Review by PicoCentauri — COMMENTED** [review]
Very nice changes. Some comments on the tests and a question on the tox configuration file.

File: tox.ini (hunk 6 above)
Comment (Luthaf) [reply tox.ini:17]: because we (i.e. tox, with the `package=env` option) install with `--no-deps`

Comment (PicoCentauri) [reply tox.ini:17]: ahh okay, maybe lets add a comment line there that I am will not ask again the next time we touch this file.

File: python/tests/log.py (hunk 2 above)
Comment (Luthaf) [reply python/tests/log.py:22]: yeah, the log testing options in pytest look powerful but also a bit hard to understand for non-pytest experts. Given that we only test a handful of cases, I find something more explicit to be easier to follow

**Review by PicoCentauri — APPROVED** [review]
🚢
